Starting a subscription correctly stops any active subscriptions for that publication

corporealfunk opened the pull request:
Fixes issue #26

corporealfunk committed:
Starting a subscription correctly stops any active subscriptions for that publication

samybob1 commented on src/subscriptions.js:
```diff
@@ -54,12 +54,12 @@ export default tracker => ({ dispatch }) => next => action => {
       const subscription = subscribe();
       const { subscriptionId } = subscription;
 
+      stopSubscription(key);
```
I think this should stay before `const subscription = subscribe();`, otherwise we are stopping the newly created subscription each time!

corporealfunk replied:
Ah, yes! agree!

samylaumonier reviewed (changes requested):
Thank you @corporealfunk, you are right there is a bug!

corporealfunk committed:
Moving stopSubscription() call before subscribe() call

corporealfunk committed:
Updating tests to reflect the correct stopping of subscriptions during subscription start

corporealfunk commented:
I just updated the pull request and updated the tests to reflect the fixed functionality (mock snapshots had to reflect that we do in fact stop a subscription when it was previously active).

samybob1 commented:
Sounds good, thank you!
